framework/include/eventloop/eventloop.h

```diff
@@ -0,0 +1,105 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/**
+ * @defgroup Eventloop Eventloop
+ * @ingroup Eventloop
+ * @brief Provides APIs for Eventloop
+ * @{
+ */
+/**
+ * @file eventloop/eventloop.h
+ */
+#ifndef __EVENTLOOP_H__
+#define __EVENTLOOP_H__
+
+#include <libtuv/uv.h>
+
+/**
+ * @brief Error Type of Result Value returned from Event Loop
+ * @details If operation is failed, these defined values will be returned.
+ */
+enum el_result_error_e {
+	EVENTLOOP_INVALID_PARAM = -1,
+	EVENTLOOP_BUSY = -2,
+	EVENTLOOP_OUT_OF_MEMORY = -3,
+};
+
+/**
+ * @brief Event Loop Timer structure
+ */
+typedef uv_timer_t event_timer;
+/**
+ * @brief Event Loop Timeout Callback
+ */
+typedef void (*timeout_callback)(void *data);
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+/**
+ * @brief Create timer to call the function after the given time
+ * @details @b #include <eventloop/eventloop.h>
+ * @param[in] timeout milliseconds until callback fires
+ * @param[in] func the callback function to be called
+ * @param[in] data data to pass to func when func is called
+ * @return On success, OK is returned. On failure, defined negative value is returned
+ * @since TizenRT v2.0 PRE
+ */
+int eventloop_add_timeout(unsigned int timeout, timeout_callback func, void *data);
```

Comment:
Adding timeout is strange for me.
This api might be implemented by eventloop_add_timer with interval=0.

Reply:
Yes, right. but we should seperate timeout and timer because of cleaning resources.
In case of timeout, some resources like timer_cb_t should be freed after execution once.
Otherwise, these for timer are freed when user calls eventloop_delete_timer with timer pointer got from eventloop_add_timer.